doc/arkode/guide/source/Mathematics.rst

```diff
@@ -2074,3 +2074,51 @@ step attempt, or fails with the minimum step size, then the integration is halte
 and an error is returned. In this case the user may need to employ other
 strategies as discussed in :numref:`ARKODE.Usage.ARKStep.Tolerances` and
 :numref:`ARKODE.Usage.ERKStep.Tolerances` to satisfy the inequality constraints.
+
+.. _ARKODE.Mathematics.Relaxation:
+
+Relaxation Methods
+==================
+
+For cases where the problem :eq:`ARKODE_IVP` is conservative or dissipative with
+respect to a smooth convex function :math:`\xi(y(t))`, it is desirable to have
+the numerical method preserve these properties. That is
+:math:`\xi(y_n) = \xi(y_{n-1}) = \ldots = \xi(y_{0})` for conservative problems
+and :math:`\xi(y_n) \leq \xi(y_{n-1})` for dissipative problems. To this end,
+ARKODE supports relaxation methods
+:cite:p:`ketcheson2019relaxation, kang2022entropy, ranocha2020relaxation` to
+ensure dissipation or preservation of a global function.
+
+Relaxation methods compute a new solution
+
+.. math::
+   y_r = y_{n-1} + r d = r y_n + (1 - r) y_{n - 1}
+   :label: ARKODE_RELAX_SOL
+
+where :math:`d \equiv h_n \sum_{i=1}^{s}(b^E_i f^E_i + b^I_i f^I_i)` is the update
```

Comment:
I believe this will work for a time-dependent mass matrix because we store `M(t)^{-1} f(t, y)` but will not currently work for a fixed mass matrix (because we store `f(t,y)` not `M^{-1} f(t,y)`). I've updated the `delta_y` and `delta_e` functions to compute `M^{-1} d` where `d` is the update to `y_n` and `M^{-1} f(t, y)`.

Reply:
I saw that you added some commits to support mass matrices.  Do you have any tests to explore whether this works?  If you don't have a test problem that can do this, then how about adding a comment here in the documentation akin to "We have not yet tested use of relaxation methods for problems with non-identity mass matrices, so feedback from users who explore this combination would be appreciated."